Source/SwiftLintFramework/Rules/RedundantVoidReturnRule.swift

```diff
@@ -41,7 +42,7 @@ public struct RedundantVoidReturnRule: ASTRule, ConfigurationProviderRule, Corre
         ]
     )
 
-    private let pattern = "\\s*->\\s*(?:Void|\\(\\s*\\))"
+    private let pattern = "\\s*->\\s*(?:Void(?=\\b)|\\(\\s*\\))"
```

Comment:
you can remove the lookahead to make it simpler: `"\\s*->\\s*(?:Void\\b|\\(\\s*\\))"`